### DISCUSSION TITLE
Building docker IDE based on "theia-apps" with fresh VS-code extensions

### DISCUSSION THREAD

**Discussion by ratermir** [eclipse-theia/theia#10739] [Q&A] [answered]
I sucesfully built IDE based on https://github.com/theia-ide/theia-apps (python example), but I found there are old VS-code extensions (for example ms-python is from year 2020), and I can't update them to latest versions. In the extension list there are only listed the old ones, even if after clicking on title it in the IDE, I am correctly navigated to to latest version of the extension (for exmple https://open-vsx.org/extension/ms-python/python), after installing it, the 2020 version is installed. 

I am familiar with docker and python, not too much with NodeJS / javascript (even if some experiences I have) so I'd appreciate some "stupid" instruction how to update the application. I put it here since theia-apps seems not to be too live and there is no discussion there.

**Comment by vince-fugnitto** [comment, marked as answer]
@ratermir thank you for the discussion :+1: 

Given you have your own docker image and `package.json` following `theia-python-docker`, you'll want to update the `package.json` which ultimately represents the theia application. In the `package.json` you'll notice a field [`theiaPlugins`](https://github.com/theia-ide/theia-apps/blob/master/theia-python-docker/latest.package.json#L37-L103) which represents which plugins will be included by default in your application. You can update this list (the urls to point to newer versions) as you need but you should ensure that the extensions are compatible.

The framework uses [open-vsx](https://open-vsx.org/) as an open-source alternative to the vscode marketplace.
